### PR TITLE
BFB-375: Remove leaderboard folder

### PR DESCRIPTION
**Pull request by m4rtint**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-375

**Description**
- Remove leaderboardd and all big items associated with it if needed

**Screenshots / GIFs**
*[insert here]*

### CODE REVIEW

File: Assets/_BForBoss/_Core/Scripts/Managers/RaceCourseWorldManager.cs [hunk 1]
```diff
@@ -1,65 +0,0 @@
-using Perigon.Analytics;
-using Perigon.Leaderboard;
-using Perigon.Utility;
-using Sirenix.OdinInspector;
-using UnityEngine;
-using UnityEngine.Rendering;
-
-namespace BForBoss
-{
-    public class RaceCourseWorldManager : BaseWorldManager
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Core/Scripts/Managers/RaceCourseWorldManager.cs:10]: Is this really not being used by any of the scenes atm?

File: Assets/_BForBoss/_Core/Scripts/Managers/WeaponRaceCourseWorldManager.cs [hunk 2]
```diff
@@ -1,82 +0,0 @@
...
-using Perigon.Analytics;
-using Perigon.Leaderboard;
-using Perigon.Utility;
-using Sirenix.OdinInspector;
-using UnityEngine;
-using UnityEngine.Rendering;
-
-namespace BForBoss
-{
-    public class WeaponRaceCourseWorldManager: BaseWorldManager
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Core/Scripts/Managers/WeaponRaceCourseWorldManager.cs:11]: This being used in `WeaponRaceCourse.unity`, `MS2_RaceCourse.unity`, `MatTestScene.unity`

File: Assets/_BForBoss/_Core/Scripts/Managers/RaceCourseWorldManager.cs (hunk 1 above)
Comment (Gursaanj) [reply Assets/_BForBoss/_Core/Scripts/Managers/RaceCourseWorldManager.cs:10]: it is being used in `MS1_RaceCourse.unity`

**Review by Gursaanj — COMMENTED** [review]
Looks like the world manager scripts being deleted affect some scenes (that I think should just be deleted too IMO) but if this goes to main with making the appropriate changes to the listed scenes , the game will definitely not be playable or scene switchable

File: Assets/_BForBoss/_Core/Scripts/Managers/RaceCourseWorldManager.cs (hunk 1 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Core/Scripts/Managers/RaceCourseWorldManager.cs:10]: Yeah I made the change in the other pr to only show the wave sceheu

File: Assets/_BForBoss/_Core/Scripts/Managers/WeaponRaceCourseWorldManager.cs (hunk 2 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Core/Scripts/Managers/WeaponRaceCourseWorldManager.cs:11]: All the race course senes are removed from the scene selection.

**Comment by m4rtint** [comment]
> Looks like the world manager scripts being deleted affect some scenes (that I think should just be deleted too IMO) but if this goes to main with making the appropriate changes to the listed scenes , the game will definitely not be playable or scene switchable

I'm just deleting them one at a time. So it's more readable what has been deleted. This is only part of it. 

The race course scenes are not in the scene switcher anyway, so it's fine for now. 

I'm splitting the deleting into separate PR's

**Review by Gursaanj — APPROVED** [review]

**Pull request merged**